src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = ({ ...state });
+  const arrayFromState = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const extraDatavalue = actions[i].extraData;
+    const typeOfAction = actions[i].type;
+    const removeValue = actions[i].keysToRemove;
+
+    switch (typeOfAction) {
+      case 'addProperties':
+        for (const newProperty in extraDatavalue) {
+          newState[newProperty] = extraDatavalue[newProperty];
+        }
+        arrayFromState.push({ ...newState });
+        break;
+
+      case 'removeProperties':
+        for (let j = 0; j < removeValue.length; j++) {
+          delete newState[removeValue[j]];
+        }
+        arrayFromState.push({ ...newState });
+        break;
+
+      default:
+        Object.keys(newState).forEach(key => delete newState[key]);
+        arrayFromState.push({ ...{} });
```

Comment:
You do arrayFromState.push({ ...newState }); in every case, you can do it after all cases.